### ISSUE TITLE
Option to turn off JIT image loading

### ISSUE THREAD

**Issue by halcyon1234** [apxltd/what-bugs#80]
I abso-fucking-loutly cannot stand scrolling down, and seeing images "fade in" when I get to them.  Fuck that, fuck that noise, fuck it hard.

Please provide a profile toggle to turn that crap off.  Load the images when I request the page (kinda almost sort EXACTLY like websites are supposed to do...)

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix:
// ==UserScript==
// @name        tdwtf forum - unfuck image loading
// @namespace   Lorne
// @include     https://what.thedailywtf.com/topic/*
// @version     1
// @grant       none
// @runat       document-end
// ==/UserScript==

var $ = unsafeWindow.jQuery;

$(document).ready(function()
{

//    $(".img-responsive").bind("changeData": function() {console.log("data was changed"); } );
  //  $(".img-responsive").bind("setData": function() {console.log("data was setData"); });
    LoadImagesForce();

});

function LoadImagesForce()
{
    var $ = unsafeWindow.jQuery;
    var $imgs = $(".img-responsive");

   // console.log($imgs);

```
$imgs.each(function()
{
    var $ = unsafeWindow.jQuery;
    var $this = $(this);
   // console.log(this);
   // console.log("this.src", $this.attr("src"));
   // console.log("this.data.src", $this.data("src"));
    //console.log("this.data.state", $this.data("state"));        
    if($this.attr("src")=="about:blank")
    {
       // console.log("loading ", this);

        $this.data("state", "loaded");
        $this.attr("data-state", "loaded");
        $this.attr("src", $this.data("src"));
       // console.log("loaded  ", this);

    }
        //console.log('$this.data("src") = ', $this.data("src"));
        //console.log('$this.attr("src") = ', $this.attr("src"));

});

setTimeout(LoadImagesForce, 3000);
```

}

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix:
// ==UserScript==
// @name        tdwtf forum - unfuck image loading
// @namespace   Lorne
// @include     https://what.thedailywtf.com/topic/*
// @version     1
// @grant       none
// @runat       document-end
// ==/UserScript==

var $ = unsafeWindow.jQuery;

$(document).ready(function()
{

//    $(".img-responsive").bind("changeData": function() {console.log("data was changed"); } );
  //  $(".img-responsive").bind("setData": function() {console.log("data was setData"); });
    LoadImagesForce();




});

function LoadImagesForce()
{
    var $ = unsafeWindow.jQuery;
    var $imgs = $(".img-responsive");
    
   // console.log($imgs);
    
    $imgs.each(function()
    {
        var $ = unsafeWindow.jQuery;
        var $this = $(this);
       // console.log(this);
       // console.log("this.src", $this.attr("src"));
       // console.log("this.data.src", $this.data("src"));
        //console.log("this.data.state", $this.data("state"));        
        if($this.attr("src")=="about:blank")
        {
           // console.log("loading ", this);

            $this.data("state", "loaded");
            $this.attr("data-state", "loaded");
            $this.attr("src", $this.data("src"));
           // console.log("loaded  ", this);
            
        }
            //console.log('$this.data("src") = ', $this.data("src"));
            //console.log('$this.attr("src") = ', $this.attr("src"));

    });
    
    setTimeout(LoadImagesForce, 3000);
}

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix:

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
I'm not enjoying having to write userscripts to turn off stupid behavior to make standard browser things work like standard browser things.  Stop fucking around with img tags like this.

Fix via userscript attached.

**Comment by halcyon1234** [comment]
wow sorry about all the dupes I clicked comment and literally nothing happened-- no postback, no grey out comment, no auto-stream in.  Fucking github.

**Comment by sjennison** [comment]
The reason this feature was added was to reduce the jellypotato effect of having images load after the page has already jumped to the target post... I've reliably had it hit within a few posts now, rather than almost never get close like before.

What exactly is annoying about this behavior? It helps a lot with jellypotato and probably with mobile bandwidth so I can't really see a reason you'd want to turn it off...

**Comment by halcyon1234** [comment]
I'm not on mobile.

It's annoying for two reasons:

1) I'll be scrolling, while scanning text.  Suddenly, in the bottom of the screen/corner of my eye BAM huge motherfucking image appears and does fancy-pants fade in.  It's distracting fluff that I don't need.

2) It CAUSES jellypotato.  I'll scroll down.  It won't load the images until AFTER I've scrolled past them.  So I see a block of text with absolutely no indication there is an image there.  Remember, the image is 2px x 2px with no border.  So after I've scrolled past the post-- missing more than half the content because the image isn't there-- suddenly the image fades in-- possibly offscreen-- an jp's the thread around. 

3) (Yes I know I said two but I thought of more) this is putting in extra work to break a browser feature, just to fix it, and do it wrong.  All IMG tags have their src set.  So browsers should be requesting all those images during page render.  But instead there's a race for this janky piece of javascript to kick in, to remove the src attribute before the image loads.  Only to put it right back in later on-- at the wrong time-- and break how images work.

So it doesn't help with bandwidth, or jellypotatoe, or making the site more usable.
